### PR TITLE
preview support for repository visibility

### LINKED ISSUE #2149: preview support for repository visibility
- https://developer.github.com/v3/repos/ (several endpoints)

### PR DESCRIPTION
**Pull request by IvanJosipovic**
Fixes #2149

### CODE REVIEW

**Comment by cliffchapmanrbx** [comment]
It's been a few months, any update on this?

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public class RepositoryRequest : RequestParameters
         /// <value>
         ///  The visibility.
         /// </value>
-        public RepositoryVisibility? Visibility { get; set; }
+        public RepositoryRequestVisibility? Visibility { get; set; }
```
Comment (shiftkey) [inline Octokit/Models/Request/RepositoryRequest.cs:46]: Noting this for the release notes: breaking change `RepositoryVisibility` -> `RepositoryRequestVisibility`

**Review by shiftkey — APPROVED** [review]
Thanks for the contribution @IvanJosipovic! I'm going to pull this in to the next update.

**Comment by shiftkey** [comment]
release_notes: Add support for repository visibility preview

**Comment by shiftkey** [comment]
advisories: due to supporting repository previews, the type on `RepositoryRequest.Visibility` has been renamed from `RepositoryVisibility` to `RepositoryRequestVisibility` to avoid clashing with the response type

**Pull request merged**